src/main/java/io/appform/nautilus/funnel/graphmanagement/ESEdgeBasedGraphBuilder.java

```diff
@@ -159,14 +160,29 @@ public Graph build(final String tenant, Context analyticsContext, GraphRequest g
                 }
             }
 
+            ImmutableList<FlatPath> paths = flatPathListBuilder.build();
+
+
+            PathUtils.rankNodes(paths.stream().map(flatPath -> flatPath.getPath()).collect(Collectors.toCollection(ArrayList::new)))
```

Comment:
Move to two separate lines. Code is not readable. In general whenever using chaining, try to keep links on separate lines, properly indented. So that the code is readable.